File: SimpleSchedulerDomainModels/JobWithWorkerID.cs
namespace SimpleSchedulerDomainModels;

public record class JobWithWorkerID(
    long ID,
    long ScheduleID,
    DateTime InsertDateUTC,
    DateTime QueueDateUTC,
    DateTime? CompleteDateUTC,
    string StatusCode,
    Guid AcknowledgementCode,
    DateTime? AcknowledgementDate,
    bool HasDetailedMessage,
    string? FriendlyDuration,
    long WorkerID,
    string WorkerName
) : Job(
    ID,
    ScheduleID,
    InsertDateUTC,
    QueueDateUTC,
    CompleteDateUTC,
    StatusCode,
    AcknowledgementCode,
    AcknowledgementDate,
    HasDetailedMessage,
    FriendlyDuration
);

File: SimpleSchedulerCommandLine/models/jobwithworkerid.go
package models

type JobWithWorkerID struct {
	ID                  int64       `json:"id"`
	ScheduleID          int64       `json:"scheduleID"`
	InsertDateUTC       *CustomTime `json:"insertDateUTC"`
	QueueDateUTC        *CustomTime `json:"queueDateUTC"`
	CompleteDateUTC     *CustomTime `json:"completeDateUTC"`
	StatusCode          string      `json:"statusCode"`
	AcknowledgementCode string      `json:"acknowledgementCode"`
	AcknowledgementDate *CustomTime `json:"acknowledgementDate"`
	HasDetailedMessage  bool        `json:"hasDetailedMessage"`
	FriendlyDuration    string      `json:"friendlyDuration"`
	WorkerID            int64       `json:"workerID"`
	WorkerName          string      `json:"workerName"`
}
